<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1355</width>
    <height>799</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QCustomPlot" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>40</y>
      <width>641</width>
      <height>621</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="genRandomGraphBtn">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>200</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::PreventContextMenu</enum>
    </property>
    <property name="text">
     <string>Случайный граф</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="matrixSpinBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>450</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>80</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QSlider" name="matrixSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>450</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="automorphismBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>200</y>
      <width>391</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Автоморфизм</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="outputEdit">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>580</y>
      <width>321</width>
      <height>151</height>
     </rect>
    </property>
    <property name="tabStopWidth">
     <number>74</number>
    </property>
    <property name="textInteractionFlags">
     <set>Qt::TextEditorInteraction</set>
    </property>
   </widget>
   <widget class="QLabel" name="outputLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>550</y>
      <width>231</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Автоморфизмы:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="shuffleBtn">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>240</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Перемешать</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="matrix_2Edit">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>360</y>
      <width>271</width>
      <height>151</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="isomorphismBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>240</y>
      <width>391</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Изоморфизм (требуется ввести обе матрицы)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gomomorphismBtn">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>280</y>
      <width>391</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Гомоморфизм (требуется ввести обе матрицы)</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="matrix_2SpinBox">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>80</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>450</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
   </widget>
   <widget class="QSlider" name="matrix_2Slider">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>120</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>450</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QTextEdit" name="matrixEdit">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>360</y>
      <width>271</width>
      <height>151</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="matrix_2Label">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>330</y>
      <width>231</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Матрица 2</string>
    </property>
   </widget>
   <widget class="QLabel" name="matrixLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>330</y>
      <width>231</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Матрица 1</string>
    </property>
   </widget>
   <widget class="QSlider" name="rangeSlider">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>90</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>450</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="rangeSpinBox">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>50</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>450</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
   </widget>
   <widget class="QLabel" name="matrixLabel_2">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>10</y>
      <width>231</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Диапазон (от 0 до x):</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>520</y>
      <width>661</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>653</x>
      <y>0</y>
      <width>21</width>
      <height>761</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1355</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>matrixSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>matrixSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>68</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>matrixSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>matrixSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>31</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>30</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>matrix_2SpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>matrix_2Slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>matrix_2Slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>matrix_2SpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rangeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>rangeSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rangeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>rangeSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>582</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
